src/box/lua/schema.lua

```diff
@@ -2362,12 +2405,16 @@ base_index_mt.pairs_ffi = function(index, key, opts)
     local ibuf = cord_ibuf_take()
     local pkey, pkey_end = tuple_encode(ibuf, key)
     local itype = check_iterator_type(opts, pkey + 1 >= pkey_end);
+    local svp = builtin.box_region_used()
+    local after = check_pairs_opts(opts)
```

Comment:
`check_pair_opts` should check and return *all* options passed to `pairs`, just like `check_select_opts`.

Reply:
Added iterator type check to this function